src/js/PSVPromise.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Promise extension and helper
+ * @param {Function} resolver
+ * @constructor
+ */
+function PSVPromise(resolver) {
+
+  var gResolve;
+
+  var promise = new Promise(function(resolve, reject) {
+    // Send args like normal promise
+    gResolve = resolution(resolve);
+    resolver(resolution(resolve), resolution(reject));
+  });
+
+  promise.cancel = function() {
+    setTimeout(function() {
+      gResolve();
+    }, 500);
```

Comment:
why is the timeout needed ?

Reply:
It's a precaution in case cancel is called right away synchronously. I didn't test without it though, let me test with removing it.

Reply:
not needed